atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/PersistentLockManager.java

```diff
@@ -62,25 +66,32 @@ public synchronized void shutdown() {
         log.info("Shutdown completed!");
     }
 
-    public synchronized void acquirePersistentLockWithRetry() {
+    // We don't synchronize this method, to avoid deadlocking if {@link #shutdown} is called while attempting
+    // to acquire a lock.
+    public void acquirePersistentLockWithRetry() {
```

Comment:
see comment below, but feels like we should either throw if client calls this and we're shutdown, or make this method return false indicating they didn't actually acquire the lock